spec/lib/appsignal/hooks/active_support_event_reporter_spec.rb

```diff
@@ -0,0 +1,53 @@
+describe Appsignal::Hooks::ActiveSupportEventReporterHook do
+  begin
+    require "active_support/event_reporter"
+  rescue LoadError
+  end
+
+  describe "#dependencies_present?" do
+    subject { described_class.new.dependencies_present? }
+
+    if DependencyHelper.rails8_1_present?
+      context "when ActiveSupport::EventReporter is present" do
+        it { is_expected.to be_truthy }
+      end
+    else
+      context "when ActiveSupport::EventReporter is not present" do
+        it { is_expected.to be_falsy }
+      end
+    end
+  end
+
+  if defined?(::ActiveSupport::EventReporter)
+    describe "#install" do
+      before do
+        # Mock Rails.event
+        stub_const("MockEventReporter", Class.new do
+          attr_reader :subscribers
+
+          def initialize
+            @subscribers = []
+          end
+
+          def subscribe(subscriber)
+            @subscribers << subscriber
+          end
+        end)
+
+        mock_event_reporter = MockEventReporter.new
```

Comment:
I recommend using an instance double for this, so that if the Rails API for this ever changes, our tests fail too, and it doesn't silently keep passing the test. Keeping us unaware anything is broken.
https://rspec.info/features/3-12/rspec-mocks/verifying-doubles/instance-doubles/